transaction-flow.md

```diff
@@ -432,7 +440,7 @@ Please note when sending `MAD::Cash` payments you should subscribe to the `recip
 }
 ```
 
-The `reference` can also be provided optionally for MAD::Cash, but if you want to use this functionality please contact us for more details.
+The payment reference can also be provided in the recipient details hash optionally for `MAD::Cash` in which case it will be used instead of the one we generate. The field you have to provide in the hash is called `reference`. If you wish to use this functionality, please contact us for more details.
```

Comment:
It would be nice to either provide an example or link to one here. Other than that looks good.

Reply:
The reference scheme will be different I believe for every partner we'll have (to avoid them clashing), so I don't think we'll need explicit examples here for this functionality. It's going to be demonstrated to them separately.

Reply:
That feels a bit icky to me just to say that we will contact them. I think we should have this something like:

```ruby
{
  reference: CUSTOMER_PREFIX + incrementing_number
}
```

So at least they know how to implement this. As a developer I know I would prefer this over being told the implementation details will be given if they contact us. That being said I will leave it to your judgement.